### PR TITLE
DBZ-3124 Add ENUM column type support for Vitess 9.0.0

### PR DESCRIPTION
**Pull request by keweishang**
https://issues.redhat.com/browse/DBZ-3124

Related PRs:

- Website: https://github.com/debezium/debezium.github.io/pull/641
- Vitess Connector documentation: https://github.com/debezium/debezium/pull/2142

### CODE REVIEW

File: pom.xml
Comment (keweishang) [inline pom.xml:21]: I'll update the compatibility matrix on the Debezium website as well

Comment (keweishang) [reply pom.xml:21]: Corresponding compatibility matrix PR: https://github.com/debezium/debezium.github.io/pull/641

File: src/main/java/io/debezium/connector/vitess/VitessType.java
Comment (gunnarmorling) [inline src/main/java/io/debezium/connector/vitess/VitessType.java:32]: Can we wrap this into Collections.unmodifiableList(enumValues) here?

File: src/main/java/io/debezium/connector/vitess/VitessType.java
Comment (gunnarmorling) [inline src/main/java/io/debezium/connector/vitess/VitessType.java:26]: Let's use Collections.emptyList() instead.

File: src/main/java/io/debezium/connector/vitess/VitessType.java
Comment (gunnarmorling) [inline src/main/java/io/debezium/connector/vitess/VitessType.java:48]: As per above, you then can simply do return !enumValues().isEmpty().

File: src/main/java/io/debezium/connector/vitess/VitessValueConverter.java
Comment (gunnarmorling) [inline src/main/java/io/debezium/connector/vitess/VitessValueConverter.java:44]:
```suggestion
            String commaSeparatedOptions = Strings.join(",", column.enumValues());
```

File: src/main/java/io/debezium/connector/vitess/VitessValueConverter.java
Comment (gunnarmorling) [inline src/main/java/io/debezium/connector/vitess/VitessValueConverter.java:45]: How about we add an overload to `build()` which takes the values, making usage for callers like this a bit simpler?

**Review by gunnarmorling — COMMENTED** [review]
LGTM; some small suggestions inline.

File: src/main/java/io/debezium/connector/vitess/VitessType.java
Comment (keweishang) [reply src/main/java/io/debezium/connector/vitess/VitessType.java:32]: Good point. Addressed

File: src/main/java/io/debezium/connector/vitess/VitessType.java
Comment (keweishang) [reply src/main/java/io/debezium/connector/vitess/VitessType.java:26]: Good point. Addressed

File: src/main/java/io/debezium/connector/vitess/VitessType.java
Comment (keweishang) [reply src/main/java/io/debezium/connector/vitess/VitessType.java:48]: Good point. Addressed

File: src/main/java/io/debezium/connector/vitess/VitessValueConverter.java
Comment (keweishang) [reply src/main/java/io/debezium/connector/vitess/VitessValueConverter.java:44]: fixed

File: src/main/java/io/debezium/connector/vitess/VitessValueConverter.java
Comment (keweishang) [reply src/main/java/io/debezium/connector/vitess/VitessValueConverter.java:45]: Of course. I'll create a PR to overload the `builder()` static function in the Debezium main repo.

Comment (keweishang) [reply src/main/java/io/debezium/connector/vitess/VitessValueConverter.java:45]: I added the `builder()` overload to the existing Debezium main repo [PR](https://github.com/debezium/debezium/pull/2142). Would that be ok? I could also create a separate ticket and PR.

Comment (keweishang) [reply src/main/java/io/debezium/connector/vitess/VitessValueConverter.java]: I'll wait for the Debezium PR to be merged and built use the overloaded `builder()` here.

Comment (gunnarmorling) [reply src/main/java/io/debezium/connector/vitess/VitessValueConverter.java]: Just merged the core PR :)

**Comment by gunnarmorling** [comment]
@ani-sha, just seeing that the JIRA-key-less commit message didn't trigger a build failure here. Could you look into setting up the commit message job for this (and potentially other) repo? Thx!

**Comment by gunnarmorling** [comment]
@keweishang, sorry, just noticed one more thing is missing for the docs update in core: the data type table still lists enums as unsupported.

**Comment by keweishang** [comment]
> just noticed one more thing is missing for the docs update in core: the data type table still lists enums as unsupported.

@gunnarmorling, sorry I missed that. I just created a PR for updating the documentation - data type table lists enums as supported:
https://github.com/debezium/debezium/pull/2149

**Comment by keweishang** [comment]
@ani-sha I'll push an amend-commit (a.k.a. `git commit --amend`) to modify the commit message, so that every commit would have the issue key. But here's the screenshot before I pushed my amend-commit:
image

**Review by gunnarmorling — APPROVED** [review]
LGTM now, thx! Can you merge yourself and resolve the issue once CI has passed? Thanks a lot!

**Comment by keweishang** [comment]
Sure, I'll merge the PR and resolve the issue once CI has passed.

**Pull request merged**
